.github/workflows/test.yml

```diff
@@ -2,12 +2,136 @@ name: test
 on:
   - push
   - pull_request
+
 jobs:
-  build:
+  install:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'npm'
+      - run: npm install
+
+  lint:
+    runs-on: ubuntu-latest
+    needs: install
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'npm'
+      - run: npm install
+      - run: npm run lint
+
+  typecheck:
+    runs-on: ubuntu-latest
+    needs: install
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'npm'
+      - run: npm install
+      - run: npx tsc
+
+  test-postgres:
+    runs-on: ubuntu-latest
+    needs: [install, lint, typecheck]
+    strategy:
+      matrix:
+        node-version: [14, 16, 18]
+        postgres-version: [11, 14]
```

Comment:
```suggestion
        postgres-version: [11, 14]
      fail-fast: false
```

Same in the other strategy fields. This avoids cancelling parallel jobs when there is a failure. For example, if it fails on Node.js 14, I'd still want to know if it succeeds on the other versions.